src/sdkFactory/index.ts

```diff
@@ -40,12 +40,13 @@ export function sdkFactory(params: ISdkFactoryParams): SplitIO.ICsSDK | SplitIO.
     matchingKey: getMatching(settings.core.key),
     splitFiltersValidation: settings.sync.__splitFiltersValidation,
 
-    // Callback used in consumer mode (`syncManagerFactory` is undefined) to emit SDK_READY
-    onReadyCb: !syncManagerFactory ? (error) => {
+    // Callback used to emit SDK_READY in consumer mode, where `syncManagerFactory` is undefined
+    // or only instantiates submitters, and therefore it is not able to emit readiness events.
+    onReadyCb: (error) => {
       if (error) return; // don't emit SDK_READY if storage failed to connect.
       readinessManager.splits.emit(SDK_SPLITS_ARRIVED);
```

Comment:
this one will never be a shared instance right? and when we implement new flavors like node "client side", etc this will still be true? 

Otherwise we should check about emitting a duplicated SPLITS_ARRIVED which might cause an SDK_UPDATE

Reply:
When instantiating shared clients, a new `onReadyCb` is passed to the storage which only emits SDK_SEGMENTS_ARRIVED for that shared client. Therefore, no SDK_UPDATE event will be duplicated for the main client.